lib/disability_compensation/loggers/monitor.rb

```diff
@@ -99,11 +139,81 @@ def track_526_submission_without_banking_info(user_account_uuid)
+        return JSON.parse(data) if data.is_a?(String)
+
+        nil
+      rescue JSON::ParserError
+        nil
+      end
+
       # Loops through array of ActiveModel::Error instances and formats a readable log
       def format_active_model_errors(errors)
         errors.map { |error| { "#{error.attribute}": error.type.to_s } }.to_s
       end
 
+      # Log the toxic exposure changes with metadata
+      #
+      # Submits a logging event to DataDog with minimal metadata about
+      # which toxic exposure keys were removed during submission.
+      # Uses minimal data to reduce fingerprinting risk.
+      #
+      # @param in_progress_form [InProgressForm] User's saved form data (unused, for signature compatibility)
+      # @param submitted_claim [SavedClaim::DisabilityCompensation::Form526AllClaim] The SavedClaim record (unused)
+      # @param submission [Form526Submission] The Form526Submission record
+      # @param user_uuid [String] User's UUID (unused, for signature compatibility)
+      # @param change_metadata [Hash] Hash containing removed_keys and removal flags
+      # @return [void]
+      # rubocop:disable Lint/UnusedMethodArgument
+      def log_toxic_exposure_changes(in_progress_form:, submitted_claim:, submission:,
+                                     user_uuid:, change_metadata:)
+        log_data = {
+          submission_id: submission.id,
+          completely_removed: change_metadata[:completely_removed],
+          removed_keys: change_metadata[:removed_keys]
+        }
+
+        submit_event(
+          :info,
+          'Form526Submission toxic exposure orphaned dates purged',
+          "#{self.class::CLAIM_STATS_KEY}.toxic_exposure_changes",
+          log_data
+        )
+      end
+      # rubocop:enable Lint/UnusedMethodArgument
+
+      # Calculate removed keys from toxic exposure changes
+      #
+      # Analyzes differences between save-in-progress and submitted toxic exposure data
+      # to identify which keys were removed. Filters out expected removals like
+      # 'view:' prefixed UI fields and empty hash values to reduce noise.
+      #
+      # @param in_progress_toxic_exposure [Hash] Toxic exposure data from InProgressForm
+      # @param submitted_toxic_exposure [Hash, nil] Toxic exposure data from SavedClaim
+      # @return [Hash] Metadata with completely_removed and removed_keys
+      def calculate_toxic_exposure_changes(in_progress_toxic_exposure, submitted_toxic_exposure)
+        all_removed_keys = in_progress_toxic_exposure.keys - (submitted_toxic_exposure&.keys || [])
+
+        # Filter out expected removals to reduce noise:
+        # - Empty hashes contain no meaningful data
+        removed_keys = all_removed_keys.reject do |key|
+          in_progress_toxic_exposure[key].is_a?(Hash) && in_progress_toxic_exposure[key].empty?
+        end
+
+        {
+          completely_removed: submitted_toxic_exposure.nil?,
+          removed_keys: removed_keys.sort
+        }
+      end
```

Comment:
I'm not clear where in the method keys beginning with 'view:' are removed. It appears that only empty hashes are actually filtered.